android/src/main/java/app/notifee/core/ReceiverService.java

```diff
@@ -195,6 +208,16 @@ private void onActionPressIntent(Intent intent) {
           launchActivity,
           mainComponent,
           pressActionBundle.getLaunchActivityFlags());
+
+      // Close notification drawer with new action for Android 12
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+        ContextHolder.getApplicationContext()
+          .sendBroadcast(new Intent(String.valueOf(GLOBAL_ACTION_DISMISS_NOTIFICATION_SHADE)));
+      } else {
+        ContextHolder.getApplicationContext()
+          .sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
```

Comment:
Ahh - nice - this does seem like the way to handle the transition cleanly then. Okay with the manifest max sdk version and this build version code check I like it.
Thanks google ;-)

Reply:
:) thanks. I'm going to double check this today on Android 12 then merge.